### PR TITLE
Doc to show properties handling for stream update

### LINKED ISSUE #2498: Document additive nature of property overrides on stream updates

### PR DESCRIPTION
**Pull request by ilayaperumalg**
- Add examples and notes to show how properties are handled during `stream update`

Resolves #2498

### CODE REVIEW

File: spring-cloud-dataflow-docs/src/main/asciidoc/streams-skipper.adoc [hunk 1]
```diff
@@ -96,6 +96,240 @@ $ curl http://localhost:9002/info
 {"app":{"description":"Spring Cloud Stream Log Sink Rabbit Binder Application","name":"log-sink-rabbit","version":"1.2.0.RELEASE"}}
 ----
 
+=== Force update of a Stream
+
+When upgrading a stream, the --force option can be used to deploy new instances of currently deployed applications even if no applicaton or deployment properties have changed.
```
Comment (tzolov) [inline spring-cloud-dataflow-docs/src/main/asciidoc/streams-skipper.adoc]: Looks like the stream `upgrade` and `update` are used interchangeably? Perhaps we should stick with only one? Seems like the `update` is prevailing in the rest of the doc.

File: spring-cloud-dataflow-docs/src/main/asciidoc/streams-skipper.adoc [hunk 2]
```diff
@@ -96,6 +96,240 @@ $ curl http://localhost:9002/info
...
+  "applicationProperties":
+    "spring.metrics.export.triggers.application.includes": "integration**"
+    "spring.cloud.dataflow.stream.app.label": "log"
+    "spring.cloud.stream.metrics.key": "ticktock.log.${spring.cloud.application.guid}"
+    "spring.cloud.stream.bindings.input.group": "ticktock"
+    "log.name": "mylogger"
+    "spring.cloud.stream.metrics.properties": "spring.application.name,spring.application.index,spring.cloud.application.*,spring.cloud.dataflow.*"
+    "spring.cloud.dataflow.stream.name": "ticktock"
+    "spring.cloud.dataflow.stream.app.type": "sink"
+    "foo2": "bar2"
```
Comment (tzolov) [inline spring-cloud-dataflow-docs/src/main/asciidoc/streams-skipper.adoc]: Perhaps we can use [Code block with callouts](https://asciidoctor.org/docs/asciidoc-syntax-quick-reference/#source-code) to make it easy to highlight the updated properties?

File: spring-cloud-dataflow-docs/src/main/asciidoc/streams-skipper.adoc [hunk 3]
```diff
@@ -96,6 +96,240 @@ $ curl http://localhost:9002/info
...
```
Comment (tzolov) [inline spring-cloud-dataflow-docs/src/main/asciidoc/streams-skipper.adoc]: [Code block with callouts](https://asciidoctor.org/docs/asciidoc-syntax-quick-reference/#source-code) as well

File: spring-cloud-dataflow-docs/src/main/asciidoc/streams-skipper.adoc [hunk 4]
```diff
@@ -96,6 +96,240 @@ $ curl http://localhost:9002/info
...
+    "spring.cloud.stream.bindings.input.destination": "ticktock.time"
+    "foo3": "bar4"
```
Comment (tzolov) [inline spring-cloud-dataflow-docs/src/main/asciidoc/streams-skipper.adoc]: [Code block with callouts](https://asciidoctor.org/docs/asciidoc-syntax-quick-reference/#source-code) as well

File: spring-cloud-dataflow-docs/src/main/asciidoc/streams-skipper.adoc (hunk 1 above)
Comment (ilayaperumalg) [reply spring-cloud-dataflow-docs/src/main/asciidoc/streams-skipper.adoc]: Yeah, we use `update` in the SCDF side while `upgrade` in the Skipper side. But, in some cases, the `update` fits for the cases of `upgrade` and `rollback` as well.

In this specific case, the `--force` option is for the `upgrading`

File: spring-cloud-dataflow-docs/src/main/asciidoc/streams-skipper.adoc (hunk 2 above)
Comment (ilayaperumalg) [reply spring-cloud-dataflow-docs/src/main/asciidoc/streams-skipper.adoc]: Good suggestion. I will update accordingly.

**Comment by tzolov** [comment]
LGTM

**Comment by tzolov** [comment]
Rebased and merged at 72ece9cf79d7c8d22c218f8f9a820a9dc4ab9425

**Pull request closed without merging**